pkg/reconciler/common/releases.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	mf "github.com/manifestival/manifestival"
+	"golang.org/x/mod/semver"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+const (
+	KoEnvKey = "KO_DATA_PATH"
+)
+
+var cache = map[string]mf.Manifest{}
+
+func TargetVersion(instance v1alpha1.KComponent) string {
+	switch {
+	case instance.GetSpec().GetVersion() != "":
```

Comment:
Can this ever be nil? I don't know too much about the semantics of how we parse that struct.

Reply:
Yep, when the users do not specify anything in spec.version.